tomcat.native Symlink Broken on Trusty

When using the `tomcat.native` state on Trusty, I receive the following error message is in `catalina.out`:

    SEVERE: Failed to initialize connector [Connector[HTTP/1.1-8443]]
    org.apache.catalina.LifecycleException: Failed to initialize component [Connector[HTTP/1.1-8443]]
            at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:106)
            at org.apache.catalina.core.StandardService.initInternal(StandardService.java:559)
            at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:102)
            at org.apache.catalina.core.StandardServer.initInternal(StandardServer.java:813)
            at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:102)
            at org.apache.catalina.startup.Catalina.load(Catalina.java:638)
            at org.apache.catalina.startup.Catalina.load(Catalina.java:663)
            at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
            at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
            at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
            at java.lang.reflect.Method.invoke(Method.java:606)
            at org.apache.catalina.startup.Bootstrap.load(Bootstrap.java:280)
            at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:454)
    Caused by: org.apache.catalina.LifecycleException: The configured protocol 
    [org.apache.coyote.http11.Http11AprProtocol] requires the APR/native library which is not available
            at org.apache.catalina.connector.Connector.initInternal(Connector.java:972)
            at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:102)
            ... 12 more



I believe this bug was introduced in d3a36d9cf9e02ddb2c093913423f7742a542898c while creating a workaround for [LP#1326255](https://bugs.launchpad.net/ubuntu/+source/tomcat-native/+bug/1326255).  It appears to change the target of `libtcnative-1.so` instead of symlinking to it.  Since `libtcnative-1.so` is itself a symlink to `libtcnative-1.so.0.1.29`, Salt updates the link rather than throwing an error.

Here's what `/usr/lib/{x86_64-linux-gnu/,}libtcnative-1.so` looks like with `libtcnative-1` installed, but before calling highstate:

    # ls -l /usr/lib/x86_64-linux-gnu/libtcnative-1.so
    lrwxrwxrwx 1 root root 23 Dec 25  2013 /usr/lib/x86_64-linux-gnu/libtcnative-1.so -> libtcnative-1.so.0.1.29
    # ls -l /usr/lib/libtcnative-1.so
    ls: cannot access /usr/lib/libtcnative-1.so: No such file or directory

Here's what they look like after calling highstate:

    # ls -l /usr/lib/x86_64-linux-gnu/libtcnative-1.so
    lrwxrwxrwx 1 root root 25 Apr 11 10:31 /usr/lib/x86_64-linux-gnu/libtcnative-1.so -> /usr/lib/libtcnative-1.so
    # ls -l /usr/lib/libtcnative-1.so
    ls: cannot access /usr/lib/libtcnative-1.so: No such file or directory

--- a/tomcat/native.sls
+++ b/tomcat/native.sls
@@ -14,7 +14,7 @@ include:
       - file: server_xml
 
 {% if grains.get('oscodename') == 'trusty' %}
-/usr/lib/{{grains['cpuarch']}}-linux-gnu/libtcnative-1.so:
+/usr/lib/libtcnative-1.so:
   file.symlink:
-    - target: /usr/lib/libtcnative-1.so
+    - target: /usr/lib/{{grains['cpuarch']}}-linux-gnu/libtcnative-1.so
 {% endif %}

